From 944ae5bff0c0b6124c23e7493a130e98ec7bc12e Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Mon, 18 Mar 2024 00:17:03 -0700
Subject: [PATCH 09/16] ice: Include xdp.h where required

This commit accommodates commit 680ee0456a57 ("net: invert the netdevice.h
vs xdp.h dependency") which removes the "#include <xdp.h>" C preprocessor
directive from header file "netdevice.h". This commit was merged in the
v6.6 development cycle and causes compilation failures with the v6.6 kernel,
which are listed below.

We accommodate this commit by including the "xdp.h" header file as needed
from "ice_txrx.h".

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=680ee0456a57

Resolved compilation errors:

/src/ice_txrx.h:399:22: error: field xdp_rxq has incomplete type
  399 |  struct xdp_rxq_info xdp_rxq;

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/ice_txrx.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ice_txrx.h b/src/ice_txrx.h
index 4daa554..d00c754 100644
--- a/src/ice_txrx.h
+++ b/src/ice_txrx.h
@@ -5,6 +5,7 @@
 #define _ICE_TXRX_H_
 
 #include "ice_type.h"
+#include <net/xdp.h>
 
 #define ICE_DFLT_IRQ_WORK	256
 #define ICE_RXBUF_3072		3072
-- 
2.43.0

